dedalo/dedalo_users_auth.sh

```diff
@@ -0,0 +1,64 @@
+#!/bin/sh
+
+
+users_auth () {
+
+	IFS=$'\n'
+	for auth_session in $(echo $1 | jq  --compact-output '.[]'); do
+
+		sessionid=$(echo $auth_session | jq -r '.session_id')
+
+		#Check if the sessionid is present on the unit's current sessions list
+		device_session=$(echo "$2" | grep $sessionid)
+		if [ $? -eq 0 ]; then
+
+			device_mac=$(echo $device_session | awk '{print $1}')
+			auth_type=$(echo $auth_session | jq -r '.type')
+			device_username=$(echo $device_session | awk '{print $6}')
+
+			if [ "$auth_type" = "login" ]; then
+
+				username=$(echo ${auth_session} | jq -r '.username')
+				password=$(echo ${auth_session} | jq -r '.password')
+
+				#In case of temporary session, first de authenticate the device
+				if [ "$device_username" = "temporary" ]; then
+					dedalo query logout mac ${device_mac}
+				fi
+
+				#Authenticate the device
+				dedalo query login mac ${device_mac} username "${username}" password "${password}"
+
+			elif [ "$auth_type" = "temporary" ]; then
+
+				#Check if a temporary session not already exist
+				if [ "$device_username" != "temporary" ]; then
+					sessiontimeout=$(echo ${auth_session} | jq -r '.session_timeout')
+					#Open a temporary session
+					dedalo query authorize mac ${device_mac} username "temporary" sessiontimeout ${sessiontimeout}
+
+				fi
+			fi
+		fi
+done
+}
+
+while true
+do
+	devices_list=$(dedalo query list | grep -e dnat -e temporary )
+
+	if [ $? -eq 0 ]; then
+
+		. /opt/icaro/dedalo/config
```

Comment:
Is it possible to load the file just once outside the loop?
Or this is required to avoid restarting `dedalo_users_auth.service` if the config has been changed?

Reply:
It was put here to avoid the restart of the service if the config has been changed.